src/main/drivers/pwm_output.c

```diff
@@ -254,28 +306,63 @@ void motorInit(const motorConfig_t *motorConfig, uint16_t idlePulse, uint8_t mot
         IOInit(motors[motorIndex].io, OWNER_MOTOR, RESOURCE_OUTPUT, RESOURCE_INDEX(motorIndex));
         IOConfigGPIO(motors[motorIndex].io, IOCFG_AF_PP);
         
-        motors[motorIndex].pwmWritePtr = pwmWritePtr;
         if (useUnsyncedPwm) {
             const uint32_t hz = timerMhzCounter * 1000000;
-            pwmOutConfig(&motors[motorIndex], timerHardware, timerMhzCounter, hz / motorConfig->motorPwmProtocol, idlePulse);
+            pwmAnalogueMotorHardwareConfig(&motors[motorIndex], timerHardware, timerMhzCounter, hz / motorConfig->motorPwmProtocol, idlePulse);
         } else {
-            pwmOutConfig(&motors[motorIndex], timerHardware, timerMhzCounter, 0xFFFF, 0);
+            pwmAnalogueMotorHardwareConfig(&motors[motorIndex], timerHardware, timerMhzCounter, 0xFFFF, 0);
         }
-        motors[motorIndex].enabled = true;
     }
+    
+#ifdef USE_DSHOT
+    if (pwmWritePtr == pwmWriteDigital) {
+        pwmStartDigitalOutput();
+    }
+#endif
 }
 
 bool pwmIsSynced(void) 
 {
     return pwmCompleteWritePtr != NULL;
 }
 
-pwmOutputPort_t *pwmGetMotors(void)
+pwmMotorOutput_t *pwmGetMotors(void)
 {
     return motors;
 }
 
 #ifdef USE_SERVOS
+
+static void pwmAnalogueServoHardwareConfig(pwmServoOutput_t *servo, const timerHardware_t *timerHardware, uint8_t mhz, uint16_t period, uint16_t value)
```

Comment:
It will lead to code duplication. Maybe timer may implement function with code common from motor and servo, than these functions will be reduced to one-liner...